chainer/initializer.py

```diff
@@ -14,7 +17,12 @@ def __call__(self, array):
 
         Args:
             array (numpy.ndarray or cupy.ndarray):
-                An array to be initialized by this initializer.
+                An array to be initialized by this initializer. If ``None``,
+                this method returns new array.
+            shape (tuple): Shape of a return array. If ``None``, this method
+                uses ``array.shape``.
+            xp (module): :mod:`cupy` or :mod:`numpy`. If ``None``, this method
+                uses ``cuda.get_array_module(array)``.
```

Comment:
I feel that this documentation is insufficient. This method can either initialize the given array or return a new array. In the latter case, the sentence """Initializes given array.""" and """This method destructively changes the value of array.""" are not true.
As for `Args:` section, It is probably good to specify that `shape` and `xp` are valid only if `array` is `None`.